tt_metal/include/tt_metal/buffer.hpp

```diff
@@ -60,8 +52,7 @@ void ReadFromBuffer(Buffer buffer, std::vector<uint32_t> &host_buffer, bool shar
  * @param host_buffer Host buffer to copy data into.
  * @param core_id ID of the core shard to read.
  */
-void ReadFromShard(Buffer buffer, std::vector<uint32_t> &host_buffer, const uint32_t &core_id);
-
+void ReadFromShard(BufferHandle buffer, std::vector<std::uint32_t> &host_buffer, std::uint32_t core_id);
```

Comment:
1) why don't we expose a symmetric WriteShard functionality?
2) We would like to extend this in the Mesh context to target not only core_id, but device_id. I'm wondering whether there's some consideration made to the interface of how a user targets a specific core (i.e. is it always through core_id? I think no, because in some cases we explicitly pass in core_x, core_y.

Reply:
In the Mesh context, the MeshBuffer will have a handle to a DeviceHandle (MeshDevice). In this case, now we need an API to supply the device-id, instead of core-id. 

re: write shard -- this can be handled separately from this PR but would be good to track this support somewhere.

Reply:
Had an offline discussion, I think the long-term plan is to deprecate this function in favor of something like
```cpp
void ReadFromShard(BufferHandle buffer, std::vector<std::uint32_t> &host_buffer, ShardHandle shard);
```
where `ShardHandle` is an abstraction that makes sense regardless of whether it's a single device or mesh device buffer.